### PR TITLE
picolisp: update to 21.12

### PR DESCRIPTION
**Pull request by thunder-coding**
NDK r23 removed binutils.

Anyways, is this still needed?

### CODE REVIEW

**Comment by thunder-coding** [comment]
It seems like 21.12 (latest version as of now) builds. (Note: only tested on device builds).

Let's see whether I can catch up with the new release. Also the build seems to use very low level compiler features which is weird

File: packages/picolisp/build.sh [hunk 1]
```diff
@@ -1,84 +1,79 @@
...
-	CFLAGS+=" -c $LDFLAGS $CPPFLAGS"
+termux_step_post_get_source() {
+	TERMUX_PKG_SRCDIR+="/src"
 }
 
-termux_step_make_install() {
-	cd $TERMUX_PKG_SRCDIR/
+termux_step_make() {
+	cd $TERMUX_PKG_SRCDIR
+	opt -O3 --mtriple=$TERMUX_HOST_PLATFORM -o base.bc base.ll
```
Comment (thunder-coding) [inline packages/picolisp/build.sh:19]: The `opt` binary isn't shipped by NDK. Seems like we'll need to install the mighty llvm on host

File: packages/picolisp/build.sh [hunk 2]
```diff
@@ -1,84 +1,77 @@
...
-TERMUX_PKG_VERSION=20.7.24
-TERMUX_PKG_REVISION=2
-# We use our source copies since old version snapshots are not kept on main site.
-TERMUX_PKG_SRCURL=https://github.com/termux/distfiles/releases/download/2021.01.04/picolisp_${TERMUX_PKG_VERSION}.tar.gz
-TERMUX_PKG_SHA256=72e8d394ad32a6659210ac2cc4b7bb736dc9b94c6ac8d6296506b6dfdc92f90c
-TERMUX_PKG_DEPENDS="libcrypt, openssl"
+TERMUX_PKG_VERSION=21.12
+TERMUX_PKG_SRCURL=http://archive.ubuntu.com/ubuntu/pool/universe/p/picolisp/picolisp_$TERMUX_PKG_VERSION.orig.tar.gz
+TERMUX_PKG_SHA256=a06838236b7f5b52c5d587d32d31627f73cdb9775cc02a80f2cdaedd12888c7d
+# Clang is required for compiling sysdefs.c and running the binary which is not possible while cross compiling
```
Comment (xtkoba) [inline packages/picolisp/build.sh:9]: Depending on `clang` solely for that purpose feels somewhat overkill. How about cross-compiling `sysdefs.c` and include the binary that is to be run in `postinst`?

Comment (thunder-coding) [reply packages/picolisp/build.sh]: Fixed, thanks!

**Pull request merged**